transformations/transformations/parallel_routine_dispatch.py

```diff
@@ -34,20 +61,32 @@ def process_parallel_region(self, routine, region):
             return
 
         dr_hook_calls = self.create_dr_hook_calls(
-            routine, pragma_attrs['name'],
+            routine, routine.name+":"+pragma_attrs['name'],
```

Comment:
Just a small remark for a more pythonic string composition:
```suggestion
            routine, f'{routine.name}:{pragma_attrs["name"]}',
```